File: Web/RMA-Web-Shell/projects/fincare/src/app/billing-manager/models/credit-note-account.ts
import { Transaction } from './transaction';
import { Note } from 'projects/shared-components-lib/src/lib/notes/note';
import { PeriodStatusEnum } from 'projects/shared-models-lib/src/lib/enums/period-status-enum';

export class CreditNoteAccount {
  rolePlayerId: number;
  finPayeNumber: string;
  isAuthorised: Date;
  authorisedBy: string;
  authorisedDate: Date;
  transactions: Transaction[];
  note: Note;
  amount: number;
  periodStatus: PeriodStatusEnum;
}

File: Service/Billing/RMA.Service.Billing.Contracts/Entities/CreditNoteAccount.cs
using RMA.Common.Entities;
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System;
using System.Collections.Generic;

namespace RMA.Service.Billing.Contracts.Entities
{
    public class CreditNoteAccount
    {
        public int RolePlayerId { get; set; }
        public List<int> PolicyIds { get; set; }
        public string FinPayeeNumber { get; set; }
        public bool IsAuthorised { get; set; }
        public string AuthorisedBy { get; set; }
        public DateTime? AuthorisedDate { get; set; }
        public List<Transaction> Transactions { get; set; }
        public Note Note { get; set; }
        public bool IsPaymentReAllocation { get; set; }
        public bool IsCreditNoteReAllocation { get; set; }
        public string ReAllocationOriginalFinPayeeNumber { get; set; }
        public PeriodStatusEnum PeriodStatus { get; set; } = PeriodStatusEnum.Current;
    }
}



